provider/internal/provider.go

```diff
@@ -139,7 +140,11 @@ func diffConfigIgnoreInternal(
 	diffConfig func(ctx context.Context, req provider.DiffRequest) (provider.DiffResponse, error),
 ) func(ctx context.Context, req provider.DiffRequest) (provider.DiffResponse, error) {
 	return func(ctx context.Context, req provider.DiffRequest) (provider.DiffResponse, error) {
-		delete(req.News, "__internal")
+		m := req.Inputs.AsMap()
+		delete(m, "__internal")
+		delete(m, "__pulumi-go-provider-infer")
+		delete(m, "__pulumi-go-provider-version")
+		req.Inputs = property.NewMap(m)
```

Comment:
https://github.com/pulumi/pulumi/pull/19384

Reply:
Why do we need to delete these?

Also, we can replace this with `req.Inputs = req.Inputs.Delete("__internal", "__pulumi-go-provider-infer", "__pulumi-go-provider-version")` since https://github.com/pulumi/pulumi/pull/19384 merged.

Reply:
https://github.com/pulumi/pulumi-docker-build/pull/414 fixed an issue with __internal but that regressed in https://github.com/pulumi/pulumi-go-provider/pull/298 when we added some additional internal properties. Root cause seems to be https://github.com/pulumi/pulumi-go-provider/issues/288

Re: Delete, I'll save that for a followup since updating pu/pu can become its own rabbit hole.